lib/Base/Assert.fram

```diff
@@ -0,0 +1,28 @@
+(* This file is part of DBL, released under MIT license.
+ * See LICENSE for details.
+ *)
+
+import open Types
+
+(** Abort the program with given message.
+
+  Note that this function is pure. In correct programs, this function should
+  be never called at runtime, but it can be used to appease the type-checker.
+  Use this function only in impossible match clauses or in case of gross
+  violation of function preconditions (e.g., division by zero). *)
+pub let runtimeError {type T} =
+  (extern dbl_runtimeError : String ->[] T)
+
+(** Explicitly assert, that this case is impossible *)
+pub let assertImpossible {?msg : String} () =
```

Comment:
I would probably change this to just `impossible`, even more since it's in module Assert.